addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -302,12 +310,20 @@ options.registry.gallery = options.Class.extend({
      */
     notify: function (name, data) {
         this._super(...arguments);
+        if (!this.hasAddImages) {
+            return;
+        }
         if (name === 'image_removed') {
             data.$image.remove(); // Force the removal of the image before reset
             this.mode('reset', this.getMode());
         } else if (name === 'image_index_request') {
             var imgs = this._getImages();
             var position = _.indexOf(imgs, data.$image[0]);
+            if (position === 0 && data.position === 'prev') {
+                data.position = 'last';
+            } else if (position === imgs.length - 1 && data.position === 'next') {
+                data.position = 'first';
+            }
```

Comment:
I think I'm missing something :thinking: Why is it needed now ? How it was working before ?

Reply:
Because it was broken.
E.g.:
- select first image and move it to the previous position. => It reached the position before the expected one.
- select last image and move it to the next position. => Nothing happened.